spanner/value.go

```diff
@@ -1844,6 +1850,50 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}, opts ...decodeO
 		*p = y
 	case *GenericColumnValue:
 		*p = GenericColumnValue{Type: t, Value: v}
+	case protoreflect.Enum:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if reflect.ValueOf(p).Kind() != reflect.Ptr {
+			return errNotAPointer(p)
+		}
+		if code != sppb.TypeCode_ENUM {
```

Comment:
This is fine for now. But we need to keep in mind that some user might want to decode a INT64 value into enum.

Reply:
Maybe, we should make it happen for now. To be consistent with
1. the write path, where integer can be written into ENUM column.
2. the proto and bytes compatible behavior.

What do you think?

Reply:
@Shua1 
I will add the above compatibilities as a separate PR. This will make the code change clear of why the changes were added. Otherwise it might create confusion of why Bytes were added into Proto code and vice versa.

Reply:
@Shua1 
I have added the code changes and integration tests to validate the following compatibility behaviours between **Int64 and Enum**.

1. Write Enum into Int64 column
2. Write Int64 into Enum column
3. Read Int64 column value using Enum
4. Read Enum column value using Int64

Reply:
Yes @Shua1. This behaviour is valid on our side also.